test/redisOpenAM.spec.js

```diff
@@ -0,0 +1,164 @@
+/**
+ * Unit tests for redisOpenAM module.
+ */
+
+// dependencies
+var should = require('chai').should(),
+    Oauth2Strategy = require('passport-http-bearer').Strategy,
+    BasicStrategy = require('passport-http').BasicStrategy,
+    Nock= require('nock'),
+    Redis = require('then-redis');
+
+
+// module under test
+var RedisOpenAM = require('../lib/redisOpenAM');
+
+
+// constants
+var openAMBaseURL = 'https://example.com';
+var openAMTokenPath = '/access_token'
+var openAMInfoPath = '/tokeninfo';
+var openAMUserPath = '/userinfo';
+var openAMMock = Nock(openAMBaseURL);
+var mockToken = 'db6e6138-3f53-4065-9610-25022a175516';
+
+// tests
+describe('Module redisOpenAM', function () {
+    var redisDb;
+
+    before(function configRedisDb() {
+        redisDb = Redis.createClient();
+    });
+
+    beforeEach(function setupMockOptions() {
+        redisDb.flushall();
+    });
+
+    it('should have property basicKey and be a function', function () {
+        RedisOpenAM.should.have.property('basicKey').and.be.a.Function;
+    });
+
+    it('should have property oauth2Key and be a function', function () {
+        RedisOpenAM.should.have.property('oauth2Key').and.be.a.Function;
+    });
+
+    it('should have property basic and be a function', function () {
+        RedisOpenAM.should.have.property('basic').and.be.a.Function;
+    });
+
+    it('should have property oauth2 and be a function', function () {
+        RedisOpenAM.should.have.property('oauth2').and.be.a.Function;
+    });
+
+    describe('the createBasicStrategy function', function () {
+        var basicStrategy;
+
+        before(function referenceBasicStrategy() {
+            basicStrategy = RedisOpenAM.basic({
+                openAMBaseURL: openAMBaseURL + openAMTokenPath,
+                openAMInfoURL: openAMBaseURL + openAMInfoPath,
+                scope: [ 'sub', 'username', 'email'],
+                client_id: 'client_id',
+                client_secret: 'client_secret',
+                redis: {},
+            });
+        });
+
+        it('should return an instance of BasicStrategy', function () {
+            basicStrategy.should.be.an.Object;
+            basicStrategy.should.be.an.instanceof(BasicStrategy);
+        });
+
+        describe('The verify function used by createBasicStrategy', function () {
+            var verify;
+
+            before(function getRefToVerify() {
+                verify = basicStrategy._verify;
```

Comment:
I want to unit test the verify function and this is the quickest and easiest way to do that with the least amount of mocking. However I'm open to other suggestions on testing the verify function.